### DISCUSSION TITLE
Badge over icon_Snap Card

### DISCUSSION THREAD

**Discussion by Lu1sma** [Telefonica/mistica-design#520] [Proposals]
### Context
Estamos intentando usar el componente "Badge over icon" en la dimensión de 40 x 40 px. dentro del componente "Snap card". Nos surge la problemática de que queremos usar la funcionalidad del badge para iconos dentro del "Icon wrapper", sin embargo el componente actualmente no se readapta y queda por fuera de los márgenes y dimensiones estandarizados de la snap card.

Este comportamiento del badge podría afectar a otros componentes también, como el caso de las listas. No obstante, el componente "List" usa el badge a la derecha (Parte navegable e interaccionable del componente), y no estamos seguros de si esta actualización se querría aplicar en este caso.

**Have you read the [Contributing Guidelines on Mística](https://brandfactory.telefonica.com/document/1846#/contribute/how-to-contribute)?**
- [✅] Check to confirm

**When will I receive a reply?**
Sería genial si esta actualización fuera aceptada por vuestra parte, que se incluyera en el lanzamiento de la actualización prevista en Agosto para desarrollo. Por ese motivo, lo hemos querido comentar lo antes posible.

---
**Team:** _what is your team?_

### Is your component request related to a problem? Please describe.
El problema es que el componente snap card tiene contemplado usar icon wrapper sin un badge. No podemos cubrir funcionalidades como mostrar alertas o estados al no poder incluir un badge en iconos de esta dimensión sobre el componente.

### Describe the solution you'd like and needs
Realizar una adaptación sobre la snap card en la que contemple usar el "Badge over icon" a otras dimensiones dentro del componente. No únicamente a 24 x 24 px.

### Describe alternatives you've considered
Hemos probado una solución temporal en la que utilizamos la variant de la imagen en vez de la del icono para engañar al componente y que de esta manera funcione visualmente. Sin embargo, no es una solución definitiva ya que para desarrollo no va a poder tratemos lo que sería un icono como una imagen. Como resultado puede generar un problema en el uso del framework para este caso en concreto.

Muchísimas gracias de antemano.

Luis María.

<img width="723" alt="Captura de pantalla 2022-06-27 a las 12 14 34" src="https://user-images.githubusercontent.com/94112294/175918912-2337e6e3-582b-41c5-aa79-ff7eb244c5cd.png">

**Comment by yceballost** [comment]
Buenas!

Tenemos abierta discusión aquí
https://github.com/Telefonica/mistica-design/discussions/489

**Reply by yceballost** [reply]
Estamos valorando añadirle el badge que hablamos, si se lo añadimos al iconWrapper automaticamente podrá ser usado en cualquier componente que use iconWrapper ＾◡＾

**Comment by Lu1sma** [comment]
Perfecto, Yayo. Gracias por tener en cuenta y haber entrado a valorar/discutir nuestras peticiones. 

Saludos,

**Comment by aweell** [comment]
Buenas @Lu1sma,

Con el objetivo de tener una visión de mayor amplitud sobre la necesidad, podrías proporcionarnos algunos contextos donde se utilizaría este badge como elemento de alerta o estado? Gracias!
